File: src/Swift/task_3/numbers.swift
import Foundation

// Функция для вычисления суммы и произведения цифр числа
func sumAndProductOfDigits(of number: Int) -> (sum: Int, product: Int) {
    var n = number
    var sum = 0
    var product = 1

    while n > 0 {
        let digit = n % 10
        sum += digit
        product *= digit
        n /= 10
    }

    return (sum, product)
}

// Читаем строку ввода
guard let inputLine = readLine() else { fatalError("Failed to read the line.") }
let components = inputLine.split(separator: " ")

// Первый компонент - количество чисел
guard let count = Int(components.first ?? "") else { fatalError("Invalid input") }
var result = [String]()

// Обрабатываем каждое число и проверяем условие
for i in 1..<components.count {
    guard let number = Int(components[i]) else { continue }
    let (sum, product) = sumAndProductOfDigits(of: number)
    if sum < product {
        // Если условие выполнено, добавляем индекс в результат
        result.append(String(i))
    }
}

// Выводим результат
print(result.joined(separator: " "))

File: src/Rust/task_3/numbers.rs
use std::io;

fn sum_and_product_of_digits(digits: &str) -> (u32, u32) {
    let mut sum = 0;
    let mut product = 1;
    for digit in digits.chars() {
        if let Some(d) = digit.to_digit(10) {
            sum += d;
            product *= d;
        }
    }
    (sum, product)
}

fn main() {
    let mut input = String::new();
    io::stdin().read_line(&mut input).expect("Failed to read line");

    let numbers: Vec<&str> = input.trim().split_whitespace().collect();
    let count: u32 = numbers[0].parse().expect("Please type a number!");

    for (i, &num_str) in numbers.iter().enumerate().skip(1) {
        if i > count as usize { break; } // Для случая, если введено больше чисел, чем указано.
        
        let (sum, product) = sum_and_product_of_digits(num_str);
        
        if sum < product {
            print!("{} ", i); 
        }
    }
    println!(); // Добавляем новую строку в конце вывода
}